### DISCUSSION TITLE
Are jsql parser versions backward compatible ?

### DISCUSSION THREAD

**Discussion by ashubrads** [JSQLParser/JSqlParser#1133] [Q&A]
Are jsql parser versions backward compatible? 

I have a project heavily dependent on jsql parser and using version 0.9.5 and we are re-writing the same project with an architecture revamp and willing to use jsql parser 4. The concern we are seeing, will the version mismatch creates an issue during the migration of previous project queries (using jsql parser 0.9.5) to a new project (using jsql parser 4)?

Hoping for an immediate revert. Thanks

**Comment by wumpz** [comment]
What do you mean be backward compatible? 
1. JSqlParser 4 should parse all the sqls the older version parses. There are some small exceptions here due to the introductions of much more keywords. As far as I know, there were no tests removed.
2. There are API changes for sure, not only additions. The first that comes to mind is the bracket parsing, which can result in aliases or array processing.
3. There are a lot of grammar changes from 0.9.5 to 4. So you cannot rely on that the parse tree is exactly the same. Those changes were introduced to tackle more syntactic constructs and performance issues.

However, I think huge changes in behaviour and API are listed in the corresponding release changelogs (https://github.com/JSQLParser/JSqlParser/releases).
